### ISSUE TITLE
Field Type Abstract or ReadOnly

### ISSUE THREAD

**Issue by gautamsi** [keystonejs/keystone#1765]
To be able to work on some content not editable in admin ui (probably not easy to create custom field or the editor may not be viable) I would like to propose a field which would not be rendered as editable in admin ui when creating or editing the item. 

To render it desirably/resourcefully in detail view (still editing, no view page) we may be able to provide the renderer component without writing full custom field. This will give the data more meaning. 

Some discussion on such extensibility is already there in #1739 and #1665 

This will be especially useful to scenario when you need to have custom editing experience like a page editor using wysiwyg html page designer but that data may not be shown to the admin ui,  too complex to have out there but the view only renderer may be really useful. 

More example would be to have a field which can store data like chart values gathered and pushed using graphql or calculated by backend code,  like item view analytics over last few months. In admin ui this can be displayed nicely using custom renderer component. When there is no renderer component provided it may display sanitized html.  This will also let community provide number of useful renderer components as contrib or even part of core repo

**Comment by MadeByMike** [comment]
I'd like to solve this with custom "views" for fields

**Comment by gautamsi** [comment]
custom "views" are also defined in #1054

**Comment by MadeByMike** [comment]
I think this is now resolved with read only fields: #2258 as well as virtual fields

**Comment by ArcoMul** [comment]
For anyone coming by this issue, the current (v6.5.1) way to mark a field as readonly to do so is:

```
slug: text({
  ui: {
    itemView: {
      fieldMode: "read",
    },
  },
}),
```

Documented here: https://keystonejs.com/docs/fields/overview#common-configuration
